src/common/bsod.cpp

```diff
@@ -191,12 +198,87 @@ void general_error(const char *error, const char *module) {
     }
 }
 
+void general_error_init() {
+    __disable_irq();
+    stop_common();
+
+    jogwheel_init();
+    gui_reset_jogwheel();
+
+    //questionable placement - where now, in almost every BSOD timers are
+    //stopped and Sound class cannot update itself for timing sound signals.
+    //GUI is in the middle of refactoring and should be showned after restart
+    //when timers and everything else is running again (info by - Rober/Radek)
+
+    Sound_Play(eSOUND_TYPE_CriticalAlert);
+}
+
+void general_error_run() {
+    //cannot use jogwheel_signals  (disabled interrupt)
+    while (1) {
+        wdt_iwdg_refresh();
+        if (!gpio_get(jogwheel_config.pinENC))
+            sys_reset(); //button press
+    }
+}
+
 void temp_error(const char *error, const char *module, float t_noz, float tt_noz, float t_bed, float tt_bed) {
-    char buff[128];
-    snprintf(buff, sizeof(buff),
-        "The requested %s\ntemperature was not\nreached.\n\nNozzle temp: %d/%d\nBed temp: %d/%d",
-        module, (int)t_noz, (int)tt_noz, (int)t_bed, (int)tt_bed);
-    general_error(error, buff);
+    char text[128];
```

Comment:
Since the red screen is a kind of hard error, consider using the PNG memory for all the stuff you need to allocate here, including the texts.